Installation backend from source with pip, mxdev, cookiecutter (#1260)
plone/documentation: docs/glossary.md

    [Amazon Web Services](https://aws.amazon.com/) offers reliable, scalable, and inexpensive cloud computing services.
    Free to join, pay only for what you use.

Barceloneta
    The default theme for Plone 5.

CMS
    Content Management System

> cookiecutter

Reviewer: In this context, "Cookiecutter" is a proper noun because it is a package name and must be capitalized. However when we use or create a "cookiecutter", then it is not capitalized.

```suggestion
Cookiecutter
```

    [See Cookiecutter's documentation](https://cookiecutter.readthedocs.io/en/stable/).

cookiecutter-zope-instance
    [cookiecutter-zope-instance](https://github.com/plone/cookiecutter-zope-instance) is a cookiecutter template to create a full and complex configuration of a Zope WSGI instance.

CSS
    Cascading Style Sheets (CSS) is a stylesheet language used for describing the (most of the times visual) representation of web pages.

DigitalOcean
    [DigitalOcean, Inc.](https://www.digitalocean.com/) is an American cloud infrastructure provider headquartered in New York City with data centers worldwide.

Grunt
    The JavaScript Task Runner.
    Automates the creation and manipulation of static assets for the theme.

Less
    A dynamic stylesheet language that can be compiled into {term}`CSS` (Cascading Style Sheets).

Linode
    [Linode.com](https://www.linode.com/) is an American privately owned virtual private server provider company based in Galloway, New Jersey, United States.

mxdev
    [mxdev](https://github.com/mxstack/mxdev) [mɪks dɛv] is a utility that makes it easy to work with Python projects containing lots of packages, of which you only want to develop some.

>     It builds on top of the idea to have stable version constraints and then develop from a VCS on top of it.

Reviewer: ```suggestion
    It is designed for developers who use stable version constraints, then layer their customizations on top of that base while using a version control system.
```

>     As part of above use-case sometimes versions of the stable constraints need an override with a different (i.e. newer) version.

Reviewer: ```suggestion
    This design allows developers to override their base package constraints with a customized or newer version.
```

mxmake
    mxmake helps create development environments for Python projects based on [make](https://www.gnu.org/software/make) and [mxdev](https://github.com/mxstack/mxdev).

NFS
    [Network File System](https://en.wikipedia.org/wiki/Network_File_System).

NPM
    npm is a package manager for the JavaScript programming language.
    It is the default package manager for the JavaScript runtime environment Node.js.
    Also a registry of JavaScript packages, similar to PyPI.

pip
    pip is the package installer for Python.
    See [tool recommendations](https://packaging.python.org/en/latest/guides/tool-recommendations/) for more information.

S3
    [Amazon Web Services S3](https://aws.amazon.com/s3/).
    Object storage built to store and retrieve any amount of data from anywhere.

TTW

>     Through-The-Web, changes in the browser.

Reviewer: ```suggestion
    Through-The-Web allows editing or customizing a Plone site through a web browser.
```

> 

Reviewer: Delete extra blank lines.

```suggestion
```

Author: hm, can we sort the glossary alphabetically, it would peace my mind.

Reviewer: Line 14 sorts when rendered.

```
:sorted: true
```

Or do you mean manually sort the entries, and require developers to sort new entries going forward? If so, we should document that requirement in our contributing docs.

Author: well, lets leave it by now unsorted as it is.

Cookiecutter
    A command-line utility that creates projects from cookiecutters (project templates), e.g. creating a Python package project from a Python package project template.
    [See Cookiecutter's documentation](https://cookiecutter.readthedocs.io/en/stable/).

cookiecutter-zope-instance
    [cookiecutter-zope-instance](https://github.com/plone/cookiecutter-zope-instance) is a cookiecutter template to create a full and complex configuration of a Zope WSGI instance.

CSS
    Cascading Style Sheets (CSS) is a stylesheet language used for describing the (most of the times visual) representation of web pages.

DigitalOcean
    [DigitalOcean, Inc.](https://www.digitalocean.com/) is an American cloud infrastructure provider headquartered in New York City with data centers worldwide.

Grunt
    The JavaScript Task Runner.
    Automates the creation and manipulation of static assets for the theme.

Less
    A dynamic stylesheet language that can be compiled into {term}`CSS` (Cascading Style Sheets).

Linode
    [Linode.com](https://www.linode.com/) is an American privately owned virtual private server provider company based in Galloway, New Jersey, United States.

mxdev

>     [mxdev](https://github.com/mxstack/mxdev) [mɪks dɛv] is a utility that makes it easy to work with Python projects containing lots of packages, of which you only want to develop some.

Reviewer: ```suggestion
    [mxdev](https://github.com/mxstack/mxdev) [mɪks dɛv] is a utility that makes it easy to work with Python projects containing lots of packages, and you want to develop only some of those packages.
```

CMS
    Content Management System

cookiecutter

>     A command-line utility that creates projects from cookiecutters (project templates), e.g. creating a Python package project from a Python package project template.

Reviewer: ```suggestion
    A command-line utility that creates projects from cookiecutters (project templates), e.g. creating a Python package project from a Python package project template.
    [See Cookiecutter's documentation](https://cookiecutter.readthedocs.io/en/stable/).
```

cookiecutter-zope-instance

>     cookiecutter-zope-instance is a cookiecutter template to create a full and complex configuration of a Zope WSGI instance.

Reviewer: ```suggestion
    [cookiecutter-zope-instance](https://github.com/plone/cookiecutter-zope-instance) is a cookiecutter template to create a full and complex configuration of a Zope WSGI instance.
```

Less
    A dynamic stylesheet language that can be compiled into {term}`CSS` (Cascading Style Sheets).

mxdev

>     mxdev [mɪks dɛv] is a utility that makes it easy to work with Python projects containing lots of packages, of which you only want to develop some.\

Reviewer: ```suggestion
    [mxdev](https://github.com/mxstack/mxdev) [mɪks dɛv] is a utility that makes it easy to work with Python projects containing lots of packages, of which you only want to develop some.
```

>     It builds on top of the idea to have stable version constraints and then develop from a VCS on top of it.\

Reviewer: ```suggestion
    It builds on top of the idea to have stable version constraints and then develop from a VCS on top of it.
```

>     As part of above use-case sometimes versions of the stable constraints need an override with a different (i.e. newer) version.\

Reviewer: ```suggestion
    As part of above use-case sometimes versions of the stable constraints need an override with a different (i.e. newer) version.
```

>     https://github.com/mxstack/mxdev

Reviewer: Moved link inline.

```suggestion
```

mxmake

>     TODO mxmake glossary term description

Reviewer: From: https://github.com/mxstack/mxmake/blob/main/docs/source/index.rst

```suggestion
    mxmake helps create development environments for Python projects based on [make](https://www.gnu.org/software/make) and [mxdev](https://github.com/mxstack/mxdev).
```

Reviewer 2: Even if development of mxmake is not finished, the glossary description is fine.

    Its theme is named [Barceloneta](https://github.com/plone/plonetheme.barceloneta/).
    It is based on Twitter Bootstrap 5.
    It uses [Mockup](https://github.com/plone/mockup/) as its JavaScript stack.
    [View Mockup's patterns](https://plone.github.io/mockup/dev/).

    The other frontend is {term}`Volto`.

Slate
    [Slate.js](https://docs.slatejs.org/) is a highly customizable platform for creating rich-text editors, also known as `WYSIWYG` editors.
    It enables you to create powerful, intuitive editors similar to those you've probably used in Medium, Dropbox Paper, or Google Docs.

volto-slate

>     `volto-slate` is an interactive default text editor for Volto, developed on top of {term}`Slate` while offering enhanced WYSIWYG functionality and behavior.

Reviewer: ```suggestion
    `volto-slate` is an interactive default text editor for Volto, developed on top of {term}`Slate`, offering enhanced WYSIWYG functionality and behavior.
```

elementEditor
    A generic {term}`volto-slate` plugin architecture that can be used to create other editor interactions that follow the pattern of having a button that toggles a format (an inline element).
    It also creates a separate edit form for advanced customization of the data attached to the element.

WSGI
    The Web Server Gateway Interface (WSGI, pronounced _WIZ-ghee_) is a simple calling convention for web servers to forward requests to web applications or frameworks written in the Python programming language. 

ZEO

>     [ZEO](https://zeo.readthedocs.io/) is a client-server storage for ZODB for sharing a single storage among many clients.

Reviewer: Don't change the link, save a redirect.

```suggestion
    [ZEO](https://zeo.readthedocs.io/en/latest/) is a client-server storage for ZODB for sharing a single storage among many clients.
```

ZODB

>     [A native object database for Python](https://zodb.org/).

Reviewer: Don't change the link, save a redirect.

```suggestion
    [A native object database for Python](https://zodb.org/en/latest/).
```

Zope

>     [Zope](https://zope.readthedocs.io/) is a Python-based application server for building secure and highly scalable web applications.

Reviewer: Don't change the link, save a redirect.

```suggestion
    [Zope](https://zope.readthedocs.io/en/latest/) is a Python-based application server for building secure and highly scalable web applications.
```

    [View Mockup's patterns](https://plone.github.io/mockup/dev/). 

    The other frontend is {term}`Volto`.

WSGI

>     The Web Server Gateway Interface (WSGI, pronounced whiskey) is a simple calling convention for web servers to forward requests to web applications or frameworks written in the Python programming language. 

Reviewer: ```suggestion
    The Web Server Gateway Interface (WSGI, pronounced _WIZ-ghee_) is a simple calling convention for web servers to forward requests to web applications or frameworks written in the Python programming language. 
```
